client/state/navigation/middleware.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import { NAVIGATE } from 'state/action-types';
+
+export const navigationMiddleware = () => {
+	return ( next ) => ( action ) => {
+		if ( action.type === NAVIGATE && action.path ) {
+			page( action.path );
+		}
+
+		return next( action );
+	};
```

Comment:
fun fact!

since this is doing nothing more right now than just returning the appropriate behavior, and we only have one condition, we could streamline it with two refactorings: first we can use a ternary on the condition to decide what to return; then we can get rid of the curly brackets and make this one big function expression

```js
export const navigationMiddleware = () => next => action =>
	action.type === NAVIGATE && action.path
		? page( action.path )
		: next( action );
```